### PR TITLE
Update mapping from docid to nodeids when loading hnsw index.

### PR DESCRIPTION
**Pull request by toregge**
@baldersheim : please review
@geirst : FYI

### CODE REVIEW

File: searchlib/src/vespa/searchlib/tensor/hnsw_nodeid_mapping.cpp [hunk 1]
```diff
@@ -115,6 +116,76 @@ HnswNodeidMapping::reclaim_memory(generation_t oldest_used_gen)
...
+            auto ref = _nodeids.allocate(subspaces);
+            _refs[docid] = ref;
+            auto nodeids = _nodeids.get_writable(ref);
+            for (auto& nodeid : nodeids) {
+                nodeid = 0;
+            }
+        }
+        ++docid;
+    }
+    {
```
Comment (baldersheim) [inline searchlib/src/vespa/searchlib/tensor/hnsw_nodeid_mapping.cpp:159]: Consider making smaller methods for the different stages.

**Review by baldersheim — DISMISSED** [review]

File: searchlib/src/vespa/searchlib/test/vector_buffer_reader.h [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+#include <cassert>
+#include <cstdint>
+#include <cstring>
+#include <vector>
+
+namespace search::test
+{
+
+class VectorBufferReader {
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/test/vector_buffer_reader.h:13]: Please add class comment.

File: searchlib/src/vespa/searchlib/test/vector_buffer_writer.h [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+#pragma once
+
+#include <vespa/searchlib/util/bufferwriter.h>
+#include <vector>
+
+namespace search::test
+{
+
+class VectorBufferWriter : public BufferWriter {
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/test/vector_buffer_writer.h:11]: Please add class comment.

**Review by geirst — APPROVED** [review]
👍

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**